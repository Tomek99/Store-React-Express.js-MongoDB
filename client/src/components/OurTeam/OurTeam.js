import React, { useState } from "react";
import TeamMember from "./TeamMember/TeamMember";
import styles from "./OurTeam.module.scss";
import Member from "../../data/member.json";
import Header from "../HeaderSection/HeaderSection";
import BlurScreen from "../ProfileMenu/BlurScreen/BlurScreen";
function OurTeam() {
  const [isActive, setIsActive] = useState(false);
  function handleBlurScreen() {
    setIsActive(false);
    setShowMember();
    handleScrollBar();
  }

  const [showMember, setShowMember] = useState();
  function handleShowMember(id) {
    setShowMember(id);
    setIsActive(!isActive);
    handleScrollBar();
  }

  let elementHTML = window.document.getElementsByTagName("html")[0];
  function handleScrollBar() {
    if (!isActive) elementHTML.style.overflowY = "hidden";
    else elementHTML.style.overflowY = "scroll";
  }

  return (
    <div className={styles.OurTeam}>
      <Header firstWord="Leadership" />
      <p className={styles.aboutTeam}>
        We're led by a team who constantly questions, tinkers, and chellenges to
        unlock great creativity around every turn
      </p>
      <div className={styles.team}>
        {Member.map((item, index) => (
          <TeamMember
            key={index}
            id={index}
            item={item}
            handleShowMember={handleShowMember}
            showMember={showMember}
          />
        ))}
        {isActive ? <BlurScreen handleBlurScreen={handleBlurScreen} /> : null}
      </div>
    </div>
  );
}

export default OurTeam;
