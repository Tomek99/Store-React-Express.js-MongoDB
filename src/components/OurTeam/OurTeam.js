import React from "react";
import styles from "./OurTeam.module.scss";

function OurTeam() {
  return <div className={styles.OurTeam}>OurTeam</div>;
}

export default OurTeam;
