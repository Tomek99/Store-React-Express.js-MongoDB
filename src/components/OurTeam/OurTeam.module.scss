.OurTeam {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    color: #ccc;
    height: 350px;
}
