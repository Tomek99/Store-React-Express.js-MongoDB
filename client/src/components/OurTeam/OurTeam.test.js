import { render } from "@testing-library/react";
import OurTeam from "./OurTeam";

test("should render OurTeam component", () => {
  const { container } = render(<OurTeam />);

  expect(container).toMatchSnapshot();
});
