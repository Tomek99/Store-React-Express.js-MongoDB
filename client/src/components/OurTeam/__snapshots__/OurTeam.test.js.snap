// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OurTeam component 1`] = `
<div>
  <div
    class="OurTeam"
  >
    <header
      class="headerSection"
    >
      Leadership undefined
    </header>
    <p
      class="aboutTeam"
    >
      We're led by a team who constantly questions, tinkers, and chellenges to unlock great creativity around every turn
    </p>
    <div
      class="team"
    >
      <div
        class="TeamMember"
      >
        <div
          class="memberImg"
        >
          <img
            alt=""
            src="/images/teammember.png"
          />
          <button
            class="btnArrow"
          >
            <svg
              class="iconArrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="content"
        >
          <p
            class="name"
          >
            Emmy Hoffman
          </p>
          <p
            class="position"
          >
            President
          </p>
        </div>
      </div>
      <div
        class="TeamMember"
      >
        <div
          class="memberImg"
        >
          <img
            alt=""
            src="/images/teammember.png"
          />
          <button
            class="btnArrow"
          >
            <svg
              class="iconArrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="content"
        >
          <p
            class="name"
          >
            Emmy Hoffman
          </p>
          <p
            class="position"
          >
            Chief Talen Officer
          </p>
        </div>
      </div>
      <div
        class="TeamMember"
      >
        <div
          class="memberImg"
        >
          <img
            alt=""
            src="/images/teammember.png"
          />
          <button
            class="btnArrow"
          >
            <svg
              class="iconArrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="content"
        >
          <p
            class="name"
          >
            Emmy Hoffman
          </p>
          <p
            class="position"
          >
            Chief Marketing Officer
          </p>
        </div>
      </div>
      <div
        class="TeamMember"
      >
        <div
          class="memberImg"
        >
          <img
            alt=""
            src="/images/teammember.png"
          />
          <button
            class="btnArrow"
          >
            <svg
              class="iconArrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="content"
        >
          <p
            class="name"
          >
            Emmy Hoffman
          </p>
          <p
            class="position"
          >
            Chief Strategy Officer
          </p>
        </div>
      </div>
      <div
        class="TeamMember"
      >
        <div
          class="memberImg"
        >
          <img
            alt=""
            src="/images/teammember.png"
          />
          <button
            class="btnArrow"
          >
            <svg
              class="iconArrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="content"
        >
          <p
            class="name"
          >
            Emmy Hoffman
          </p>
          <p
            class="position"
          >
            Executive Chairman
          </p>
        </div>
      </div>
      <div
        class="TeamMember"
      >
        <div
          class="memberImg"
        >
          <img
            alt=""
            src="/images/teammember.png"
          />
          <button
            class="btnArrow"
          >
            <svg
              class="iconArrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="content"
        >
          <p
            class="name"
          >
            Emmy Hoffman
          </p>
          <p
            class="position"
          >
            Executive Chairman
          </p>
        </div>
      </div>
      <div
        class="TeamMember"
      >
        <div
          class="memberImg"
        >
          <img
            alt=""
            src="/images/teammember.png"
          />
          <button
            class="btnArrow"
          >
            <svg
              class="iconArrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="content"
        >
          <p
            class="name"
          >
            Emmy Hoffman
          </p>
          <p
            class="position"
          >
            Chief Executive Officer
          </p>
        </div>
      </div>
      <div
        class="TeamMember"
      >
        <div
          class="memberImg"
        >
          <img
            alt=""
            src="/images/teammember.png"
          />
          <button
            class="btnArrow"
          >
            <svg
              class="iconArrow"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="content"
        >
          <p
            class="name"
          >
            Emmy Hoffman
          </p>
          <p
            class="position"
          >
            President
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
